clash-lib/src/Clash/Netlist/Types.hs

```diff
@@ -472,6 +472,10 @@ data HWType
   -- ^ File type for simulation-level I/O
   deriving (Eq, Ord, Show, Generic, NFData, Hashable)
 
+annotated :: [Attr'] -> HWType -> HWType
+annotated [] t = t
+annotated attr t = Annotated attr t
```

Comment:
```suggestion
-- | Smart constructor for 'Annotated'. Wraps the given type in an 'Annotated'
-- if the attribute list is non-empty. If it is empty, it will return the given
-- 'HWType' unchanged.
annotated :: [Attr'] -> HWType -> HWType
annotated [] t = t
annotated attr t = Annotated attr t
```